src/rpdk/core/jsonutils/utils.py

```diff
@@ -132,9 +132,18 @@ def traverse(document, path_parts):
     return document, tuple(path), parent
 
 
-def _resolve_ref(sub_schema, definitions):
+def _resolve_ref(sub_schema: dict, definitions: dict, last_step: bool = False):
     # resolve $ref
-    ref = nested_lookup(REF, sub_schema)
+    ref = nested_lookup(REF, sub_schema)  # should be safe (always single value)
+    # bc sub_schema is always per paranet property
```

Comment:
nit: can you please follow the same commenting style followed in the rest of the file?